### PR TITLE
Discover code

### PR DESCRIPTION
**Pull request by montassarelbehi**

### CODE REVIEW

File: angular-beans/src/main/java/angularBeans/util/CommonUtils.java [hunk 1]
```diff
@@ -214,21 +212,43 @@ public static String obtainFieldNameFromAccessor(String getterName) {
...
+	public static <T> Boolean isNullOrEmpty(final T o) {
+		if (o == null) {
 			return true;
 		}
-		if (o instanceof String){
-			return Strings.isNullOrEmpty((String) o);
+		if (o instanceof String) {
+			return "".equals(((String) o).trim());
+		}
+		if (o.getClass().isArray()) {
```
Comment (yesbutno) [inline angular-beans/src/main/java/angularBeans/util/CommonUtils.java:196]: Why a separate `isNullOrEmpty(final byte[] o)` below when you're checking arrays here?

Comment (montassarelbehi) [inline angular-beans/src/main/java/angularBeans/util/CommonUtils.java:196]: might be cause byte is a primitive need to be tested alone. isArrays works on Objects and then when I cast o i do it as an Array of Object if isEmpty or not. is quiete like in junit when you want to assertEquals an Integer :)

Comment (montassarelbehi) [inline angular-beans/src/main/java/angularBeans/util/CommonUtils.java:196]: But I want more your feedback about the duplicate callback enum class in two packages (after enum package elimination :( )

Comment (yesbutno) [inline angular-beans/src/main/java/angularBeans/util/CommonUtils.java:196]: I will give my feedback to the Callbacks later today.
But with the `isNullOrEmpty`: The above version is broken. I already have a solution but I have to know one more thing: When is an array considered to be empty? If its length is 0 (zero)? Or if all elements are null or zero? Or...?

Comment (montassarelbehi) [inline angular-beans/src/main/java/angularBeans/util/CommonUtils.java:196]: strange! isEmpty Returns true if the list contains no elements..that's why i converted the array using asList. According to zero value, we can not consider as null in all cases that's why we haven't to handle it in this kind of method.

Comment (yesbutno) [inline angular-beans/src/main/java/angularBeans/util/CommonUtils.java:196]: asList() creates a list of the one item you pass as a parameter,  in this case your array. It took me ages to see this :)

Comment (yesbutno) [inline angular-beans/src/main/java/angularBeans/util/CommonUtils.java:196]: You could use `return Array.getLength(o) == 0;` instead.
Then you can even remove the `isNullOrEmpty(byte[])` method

**Pull request merged**